docs/architecture/adr-027-deterministic-protobuf-serialization.md

```diff
@@ -45,11 +46,25 @@ with the following additions:
    must be omitted
 4. `repeated` fields of scalar numeric types must use
    [packed encoding](https://developers.google.com/protocol-buffers/docs/encoding#packed)
-   by default.
-5. Variant encoding of integers must not be longer than needed.
+5. Varint encoding must not be longer than needed:
+    * No trailing zero bytes (in little endian, i.e. no leading zeroes in big
+      endian). Per rule 3 above, the default value of `0` must be omitted, so
+      this rule does not apply in such cases.
+    * The maximum value for a varint is `FF FF FF FF FF FF FF FF FF 01`.
+      In other words, when decoded, the highest 6 bits of the 70-bit unsigned
+      integer must be `0`. (10-byte varints are 10 groups of 7 bits, i.e.
+      70 bits, of which only the lowest 70-6=64 are useful.)
+    * The maximum value for 32-bit values in varint encoding is `FF FF FF FF 0F`
+      with one exception (below). In other words, when decoded, the highest 38
+      bits of the 70-bit unsigned integer must be `0`.
```

Comment:
I understand where these 38/70 bits come from, but I still think it's confusing to mention them here.

By the 1st sentence, 32-bit values must use 5 bytes max for encoding (except for the exception you mentioned). Do you think it's clearer to cast to a 70-bit unsigned int with 38 zeroes?

Reply:
Added a small background section that provides intuitions around the 70 bits: 20af588